src/osvr/Server/ConfigFilePaths.cpp

```diff
@@ -0,0 +1,79 @@
+// Internal Includes
+#include <osvr/Util/PlatformConfig.h>
+#include <osvr/Server/ConfigFilePaths.h>
+#include <osvr/Server/ConfigureServerFromFile.h>
+#include <osvr/Util/GetEnvironmentVariable.h>
+
+// Library/third-party includes
+#include <boost/filesystem.hpp>
+
+// Standard includes
+#include <vector>
+
+namespace osvr {
+    namespace server {
+
+        inline std::string getUserConfigDirectory() {
+            namespace fs = boost::filesystem;
+            using osvr::util::getEnvironmentVariable;
+
+            fs::path configDir;
+            std::string configSubpath = "config";
+
+#if defined(OSVR_LINUX)
+            // There's currently no great location for storing log files in the
+            // XDG system. (See the STATE proposal by Debian
+            // <https://wiki.debian.org/XDGBaseDirectorySpecification#Proposal:_STATE_directory>.)
+            // So for now, we'll store our log files in the $XDG_CACHE_HOME
+            // directory.
+            //
+            // $XDG_CACHE_HOME defines the base directory relative to which user
+            // specific non-essential data files should be stored. If
+            // $XDG_CACHE_HOME is either not set or empty, a default equal to
+            // $HOME/.cache should be used.
+            auto xdg_cache_dir = getEnvironmentVariable("XDG_CONFIG_HOME");
```

Comment:
I'm going to put this as a TODO for now. The function this is in currently only returns one directory, and this change would require it to return multiple directories.